java/cdk-environment/src/main/java/sleeper/environment/cdk/buildec2/KeyPairUtil.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2022 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package sleeper.environment.cdk.buildec2;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.util.Base64;
+import java.util.Collections;
+
+public class KeyPairUtil {
+
+    private KeyPairUtil() {
+        // Prevent instantiation
+    }
+
+    public static KeyPair generate() {
+        try {
+            KeyPairGenerator generator = KeyPairGenerator.getInstance("RSA");
+            generator.initialize(2048);
+            return generator.generateKeyPair();
+        } catch (NoSuchAlgorithmException e) {
+            throw new RuntimeException("Could not generate a keypair", e);
+        }
+    }
+
+    public static void writePrivateToFile(KeyPair pair, String fileName) {
+        try {
+            Path path = Paths.get(fileName);
+            Files.deleteIfExists(path);
+            Files.createFile(path, PosixFilePermissions.asFileAttribute(
+                    Collections.singleton(PosixFilePermission.OWNER_WRITE)));
+            Files.write(path, Collections.singletonList(KeyPairUtil.privatePem(pair)));
```

Comment:
Should be able to generate a properly formatted private PEM using this class: https://www.bouncycastle.org/docs/pkixdocs1.8on/index.html

Reply:
I didn't think it'd be worth adding an extra dependency.

Reply:
You could get rid of 2 methods and their testing - what are the downsides of adding that dependency?

Reply:
I've added it, thanks.